### PR TITLE
Validate supported install modes and check requirements on Validation state

### PR DESCRIPTION
**Pull request by Jonathas-Conceicao**

### CODE REVIEW

File: updatehub/tests/common.rs [hunk 1]
```diff
@@ -49,18 +50,24 @@ impl Default for Settings {
             download_dir: None,
             config_file: None,
             timeout: None,
+            install_modes: None,
```
Comment (otavio) [inline updatehub/tests/common.rs:53]: Couldn't this to be a `Vec<_>`?

File: updatehub/src/states/validation.rs [hunk 2]
```diff
@@ -43,11 +43,13 @@ impl StateChangeImpl for Validation {
         }
 
         // Ensure the package is compatible
+        let installation_set = context.runtime_settings.get_inactive_installation_set()?;
```
Comment (otavio) [inline updatehub/src/states/validation.rs:46]: `inactive_installation_set`

File: updatehub/tests/common.rs (hunk 1 above)
Comment (Jonathas-Conceicao) [reply updatehub/tests/common.rs:53]: We could, but then we won't be able to use the derived version of `Default` trait and would need a explicit impl for the struct.

Comment (otavio) [reply updatehub/tests/common.rs:53]: Oh, makes sense.

File: updatehub/src/update_package/mod.rs [hunk 3]
```diff
@@ -60,6 +65,17 @@ impl UpdatePackageExt for UpdatePackage {
         self.inner.supported_hardware.compatible_with(&firmware.hardware)
     }
 
+    fn validate_install_modes(&self, settings: &Settings, installation_set: Set) -> Result<()> {
+        let supported = &settings.update.supported_install_modes;
+        for object in self.objects(installation_set) {
+            let mode = object.mode().to_owned();
```
Comment (otavio) [inline updatehub/src/update_package/mod.rs:71]:
```suggestion
            let mode = object.mode();
```

**Pull request merged**